<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>Box2dPage</class>
 <widget class="QWidget" name="Box2dPage">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>1000</width>
    <height>600</height>
   </rect>
  </property>
  <property name="minimumSize">
   <size>
    <width>1000</width>
    <height>600</height>
   </size>
  </property>
  <property name="windowTitle">
   <string>Form</string>
  </property>
  <widget class="QWidget" name="horizontalLayoutWidget">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>1001</width>
     <height>532</height>
    </rect>
   </property>
   <layout class="QHBoxLayout" name="horizontalLayout">
    <property name="leftMargin">
     <number>10</number>
    </property>
    <property name="topMargin">
     <number>30</number>
    </property>
    <property name="rightMargin">
     <number>10</number>
    </property>
    <property name="bottomMargin">
     <number>7</number>
    </property>
    <item>
     <widget class="QScrollArea" name="scrollArea">
      <property name="widgetResizable">
       <bool>true</bool>
      </property>
      <widget class="QWidget" name="scrollAreaWidgetContents">
       <property name="geometry">
        <rect>
         <x>0</x>
         <y>0</y>
         <width>296</width>
         <height>1200</height>
        </rect>
       </property>
       <property name="minimumSize">
        <size>
         <width>0</width>
         <height>1200</height>
        </size>
       </property>
       <widget class="QFrame" name="titleFrame">
        <property name="geometry">
         <rect>
          <x>10</x>
          <y>10</y>
          <width>281</width>
          <height>81</height>
         </rect>
        </property>
        <property name="font">
         <font>
          <pointsize>35</pointsize>
         </font>
        </property>
        <property name="frameShape">
         <enum>QFrame::StyledPanel</enum>
        </property>
        <property name="frameShadow">
         <enum>QFrame::Raised</enum>
        </property>
        <widget class="QLabel" name="titleLabel">
         <property name="geometry">
          <rect>
           <x>10</x>
           <y>10</y>
           <width>261</width>
           <height>61</height>
          </rect>
         </property>
         <property name="font">
          <font>
           <pointsize>34</pointsize>
          </font>
         </property>
         <property name="text">
          <string>The Process</string>
         </property>
         <property name="alignment">
          <set>Qt::AlignHCenter|Qt::AlignTop</set>
         </property>
        </widget>
       </widget>
       <widget class="QLabel" name="label_3">
        <property name="geometry">
         <rect>
          <x>10</x>
          <y>100</y>
          <width>281</width>
          <height>1081</height>
         </rect>
        </property>
        <property name="font">
         <font>
          <pointsize>13</pointsize>
         </font>
        </property>
        <property name="text">
         <string>   So how does a model find the best values for &quot;m&quot; and &quot;b&quot;?  It all revolves around another type of math equation that the model uses.  This second math equation takes in the current &quot;m&quot; and &quot;b&quot; and returns a number that represents how BAD the model is at doing its job.

   For the interactive demo, we will just look at &quot;m&quot;, though &quot;b&quot; is the same process.  The graph to the right represents this kind of function that shows how bad the model is doing with the current &quot;m&quot;, and the ball represents the current &quot;m&quot; (think of the ball's X position).

   So to find the best value for &quot;m&quot;, it is the model's goal to get the ball as low as it can go.  In the real world, this involves a good deal of calculus and math.  But this calculus actually imitates how a ball would roll, even imitating the momentum of a ball!  So we can play with a ball and be doing the same thing that a real machine learning model is doing!

   Feel free to drag the ball around, and notice how the final resting place of the ball depends on where you decide to drop it.  The takeaway from this section is this:

   Our Linear Regression model's values are found by starting somewhere, and &quot;rolling down the hill&quot; of this function until we find the lowest point we can.</string>
        </property>
        <property name="alignment">
         <set>Qt::AlignJustify|Qt::AlignTop</set>
        </property>
        <property name="wordWrap">
         <bool>true</bool>
        </property>
        <property name="indent">
         <number>-1</number>
        </property>
       </widget>
      </widget>
     </widget>
    </item>
    <item>
     <widget class="canvas" name="canvasWidget" native="true">
      <property name="minimumSize">
       <size>
        <width>660</width>
        <height>495</height>
       </size>
      </property>
      <property name="maximumSize">
       <size>
        <width>660</width>
        <height>495</height>
       </size>
      </property>
     </widget>
    </item>
   </layout>
  </widget>
  <widget class="QPushButton" name="prevButton">
   <property name="geometry">
    <rect>
     <x>10</x>
     <y>560</y>
     <width>36</width>
     <height>36</height>
    </rect>
   </property>
   <property name="cursor">
    <cursorShape>PointingHandCursor</cursorShape>
   </property>
   <property name="text">
    <string/>
   </property>
   <property name="iconSize">
    <size>
     <width>48</width>
     <height>48</height>
    </size>
   </property>
  </widget>
  <widget class="QPushButton" name="nextButton">
   <property name="geometry">
    <rect>
     <x>960</x>
     <y>560</y>
     <width>36</width>
     <height>36</height>
    </rect>
   </property>
   <property name="cursor">
    <cursorShape>PointingHandCursor</cursorShape>
   </property>
   <property name="text">
    <string/>
   </property>
   <property name="iconSize">
    <size>
     <width>48</width>
     <height>48</height>
    </size>
   </property>
  </widget>
  <widget class="QLabel" name="label_2">
   <property name="geometry">
    <rect>
     <x>880</x>
     <y>560</y>
     <width>81</width>
     <height>31</height>
    </rect>
   </property>
   <property name="font">
    <font>
     <pointsize>10</pointsize>
    </font>
   </property>
   <property name="text">
    <string>Wrapping Up</string>
   </property>
  </widget>
  <widget class="QPushButton" name="randomGraphButton">
   <property name="geometry">
    <rect>
     <x>480</x>
     <y>550</y>
     <width>121</width>
     <height>31</height>
    </rect>
   </property>
   <property name="text">
    <string>Randomize Graph</string>
   </property>
  </widget>
 </widget>
 <customwidgets>
  <customwidget>
   <class>canvas</class>
   <extends>QWidget</extends>
   <header>canvas.h</header>
   <container>1</container>
  </customwidget>
 </customwidgets>
 <resources/>
 <connections/>
</ui>
